src/components/input/test/Select-test.js

```diff
@@ -425,12 +436,137 @@ describe('Select', () => {
+        { buttonContent: 'Select a value' },
+        {
+          items: ['1', '2', '3'].map(id => ({ id, name })),
+        },
+      );
+
+      const listbox = await getOpenListbox(wrapper);
+      const { width: listboxWidth } = listbox
+        .getDOMNode()
+        .getBoundingClientRect();
+
+      assert.isTrue(listboxWidth < window.window.innerWidth);
+    });
+
+    [
+      { name: 'long name'.repeat(50), shouldGrow: true },
+      { name: 'short name', shouldGrow: false },
+    ].forEach(({ name, shouldGrow }) => {
+      it('makes left-aligned Select listbox grow to the left if there is space', async () => {
+        const wrapper = createComponent(
+          { buttonContent: 'Select a value' },
+          {
+            items: ['1', '2', '3'].map(id => ({ id, name })),
+          },
+        );
+
+        const listbox = await getOpenListbox(wrapper);
+        const listboxStyleLeft = listbox.getDOMNode().style.left;
+        const listboxLeft = Number(listboxStyleLeft.replace('px', ''));
+        const { left: buttonLeft } = getToggleButton(wrapper)
+          .getDOMNode()
+          .getBoundingClientRect();
+
+        assert.equal(listboxLeft < buttonLeft, shouldGrow);
+      });
+    });
+
+    [
+      // Content is small. The listbox matches the toggle button size.
+      {
+        name: 'short name',
+        getExpectedCoordinates: (wrapper, listboxDOMNode) => {
+          const buttonDOMNode = getToggleButton(wrapper).getDOMNode();
+          const buttonLeft = buttonDOMNode.getBoundingClientRect().left;
+
+          return {
+            left: buttonLeft,
+            right: listboxDOMNode.getBoundingClientRect().width + buttonLeft,
+          };
+        },
+      },
+
+      // Content is slightly longer. The listbox matches the toggle right side
+      // but spans further than the left side
+      {
+        name: 'slightly longer name'.repeat(3),
+        getExpectedCoordinates: (wrapper, listboxDOMNode) => {
+          const buttonDOMNode = getToggleButton(wrapper).getDOMNode();
+          const buttonRect = buttonDOMNode.getBoundingClientRect();
+          const listboxRect = listboxDOMNode.getBoundingClientRect();
+          const listboxLeftIncrement = listboxRect.width - buttonRect.width;
+
+          return {
+            left: buttonRect.left - listboxLeftIncrement,
+            right: buttonRect.right,
+          };
+        },
+      },
+
+      // Content is very big, so listbox spans to the very left side and grows
+      // further than the right side of the toggle button
+      {
+        name: 'very long name'.repeat(6),
+        getExpectedCoordinates: (wrapper, listboxDOMNode) => {
+          const listboxRect = listboxDOMNode.getBoundingClientRect();
+          return {
+            left: LISTBOX_VIEWPORT_HORIZONTAL_GAP,
+            right: listboxRect.width + LISTBOX_VIEWPORT_HORIZONTAL_GAP,
+          };
+        },
+      },
+    ].forEach(({ name, getExpectedCoordinates }) => {
```

Comment:
It seems odd to me that there are an asymmetric set of tests for left and right-aligned listboxes. I would expect the behavior to be the same, but mirrored.

Reply:
Yeah, good point. They kind-of were before the last change I pushed.

I'll apply the same checks to the left-aligned Select tests, as they cover more cases.